### ISSUE TITLE
RuboCop linter gives false alarm syntax error, possibly haml lint specific

### ISSUE THREAD

**Issue by agrimm** [sds/haml-lint#251]
The following HAML code results in a false alarm about invalid Ruby syntax in the RuboCop linter:

    - choice = {required: 'mandatory'}
    = if choice[:required] == 'mandatory' then '*' end

The error given is

    Andrews-Air:sandbox_20170620 agrimm$ haml-lint page.html.haml 
    warning: parser/current is loading parser/ruby23, which recognizes
    warning: 2.3.3-compliant syntax, but you are running 2.3.1.
    warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
    page.html.haml:2 [W] RuboCop: unexpected token kEND
    (Using Ruby 2.1 parser; configure using `TargetRubyVersion` parameter, under `AllCops`)

    1 file inspected, 1 lint detected

I suspect that it isn't a RuboCop error, because running similar code in RuboCop doesn't give an error:

    Andrews-Air:sandbox_20170620 agrimm$ rubocop plain_ruby.rb 
    Inspecting 1 file
    C

    Offenses:

    plain_ruby.rb:1:10: C: Space inside { missing.
    choice = {required: 'mandatory'}
             ^
    plain_ruby.rb:1:32: C: Space inside } missing.
    choice = {required: 'mandatory'}
                                   ^
    plain_ruby.rb:2:1: C: Favor modifier if usage when having a single-line body. Another good alternative is the usage of control flow &&/||.
    if choice[:required] == 'mandatory' then '*' end
    ^^

    1 file inspected, 3 offenses detected

The Ruby code is

    choice = {required: 'mandatory'}
    if choice[:required] == 'mandatory' then '*' end

Version information:

    Andrews-Air:sandbox_20170620 agrimm$ haml-lint --version
    warning: parser/current is loading parser/ruby23, which recognizes
    warning: 2.3.3-compliant syntax, but you are running 2.3.1.
    warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
    haml-lint 0.26.0
    Andrews-Air:sandbox_20170620 agrimm$ rubocop --version
    0.49.1
    Andrews-Air:sandbox_20170620 agrimm$ ruby --version
    ruby 2.3.1p112 (2016-04-26 revision 54768) [x86_64-darwin14]

Apologies for not running under the latest master commits.

**Comment by michaelherold** [comment]
This is how the example is parsed into Ruby:

```ruby
choice = {required: 'mandatory'}
if choice[:required] == 'mandatory' then '*' end
end
```

The syntax error occurs because of the `end` that is inserted on pseudo-line 3. That `end` is inserted because the `=` operator currently treats the `if` that is passed to it as a requiring a block (see `HamlLint::RubyExtractor#visit_script` and `HamlLint::RubyExtractor#start_block_keyword?`.

We'll have to figure out a way to check whether the `end` already exists or not so we only insert it when it's needed.
